thrift/thrift0.13/src/main/java/com/linecorp/armeria/server/thrift/THttpService.java

```diff
@@ -424,6 +424,10 @@ public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exc
         req.aggregate(AggregationOptions.usePooledObjects(ctx.alloc(), ctx.eventLoop()))
            .handle((aReq, cause) -> {
                if (cause != null) {
+                   cause = Exceptions.peel(cause);
+                   if (cause instanceof HttpStatusException) {
```

Comment:
HttpResponseException as well?
Could you also fix TomecatService, JettyService and SamlService?